### PR TITLE
first draft on open source at MeiliSearch

### LINKED ISSUE #247: Define what does that mean to do open source at MeiliSearch

### PR DESCRIPTION
**Pull request by fharper**
close #247

### CODE REVIEW

**Comment by fharper** [comment]
I added all developers to review this PR since it's a small but important text. I wrote this down based on the vision who wants to have when it comes to open source at MeiliSearch. Nothing should be a surprise (even for new folks is your interviewers did a good job 🤣 )!

@react-learner you can review the ideas for now, don't copyedit until I give you the go.

**Review by qdequele — COMMENTED** [review]
I think there is some information missing. In particular: 
- What we do in open source in addition to the repos we own. Our willingness to advance other open source projects as well. 
- The fact that being open source is not necessarily incompatible with making a profit. We will release a Cloud version of MeiliSearch which will be our main source of income tomorrow. MeiliSearch will remain open source.

**Review by tpayet — APPROVED** [review]
Great for a first version, we can always come back to improve it. Love the spirit

**Review by LegendreM — APPROVED** [review]
I like this approach. 👍

**Review by CaroFG — APPROVED** [review]

**Comment by eskombro** [comment]
I think we can talk about the **MeiliSearch Sponsorship program**, which allows every MeiliSearch developer to sponsor open source projects on his/her own name, but get a refund by MeiliSearch. This is a very good example of how the company is trying to give back to the OSS community by empowering its own developers to do it for the projects they care about the most of that they want to support <3

**Review by eskombro — APPROVED** [review]
I think we can also talk about the **MeiliSearch Sponsorship program** (I'm not sure this is the 'name' of the program), which allows every MeiliSearch developer to sponsor open source projects on his/her own name or 'credit', but get a refund by MeiliSearch. This is a very good example of how the company is trying to give back to the OSS community by empowering its own developers to do it for the projects they care about the most of that they want to support ❤️

File: opensource.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+
+First, we are including the community in everything we do: **public is the default visibility of our work**. We plan the future of MeiliSearch with the help and feedback of everyone who wants to be involved. It is possible to suggest a [feature idea](https://roadmap.meilisearch.com/tabs/1-under-consideration) to influence the roadmap. We sincerely welcome pull requests, even when it's about a feature that wasn't a priority for us, as a for-profit company. Contributing doesn't only equal code: it's also about making us aware of any problems with our product by creating issues on [GitHub](https://github.com/meilisearch) or being part of our lovely [Slack community](https://slack.meilisearch.com). Note that we will have more documentation on the best ways to contribute to MeiliSearch.
+
+Obviously, we can do even more, so it's why starting from v0.21, we are moving forward with a public [specifications](https://github.com/meilisearch/specifications) process that will close the loop on involving the community and being totally transparent with everything we do product-wise.
```
Comment (curquiza) [inline opensource.md:9]: Maybe precise it's the v0.21 of MeiliSearch?

**Review by curquiza — APPROVED** [review]

**Review by nicolasvienot — APPROVED** [review]

**Review by irevoire — APPROVED** [review]

**Review by Kerollmops — APPROVED** [review]
Nice write, thank you!

**Comment by fharper** [comment]
> I think there is some information missing. In particular:
> 
> * What we do in open source in addition to the repos we own. Our willingness to advance other open source projects as well.
> * The fact that being open source is not necessarily incompatible with making a profit. We will release a Cloud version of MeiliSearch which will be our main source of income tomorrow. MeiliSearch will remain open source.

Good point! I'll update the page with this information, thanks.

**Comment by fharper** [comment]
> Great for a first version, we can always come back to improve it. Love the spirit

YES, this is exactly how we should always approach things. This is a good example: I should merge it since people agreed with what is already there, and do new PRs with improvement or part to add based on comments and how it is going in the future.

Actually, I'll add something about that specifically: we should not wait to have a final version befroe publishing things!

**Comment by fharper** [comment]
> I think we can also talk about the **MeiliSearch Sponsorship program** (I'm not sure this is the 'name' of the program), which allows every MeiliSearch developer to sponsor open source projects on his/her own name or 'credit', but get a refund by MeiliSearch. This is a very good example of how the company is trying to give back to the OSS community by empowering its own developers to do it for the projects they care about the most of that they want to support ❤️

Yes, I can't believe I didn't think about adding this. Thanks Sam!

File: opensource.md (hunk 1 above)
Comment (fharper) [reply opensource.md:9]: You mean for everything? I specified it for the specifications part.

**Comment by fharper** [comment]
> Nice write, thank you!

Yes, CTO approved 🤣

**Comment by fharper** [comment]
> I like this approach. 👍

Glad you like it!

File: opensource.md (hunk 1 above)
Comment (curquiza) [reply opensource.md:9]: I meant this sentence:
> Obviously, we can do even more, so it's why starting from v0.21

I personally get the sentence, I'm not sure an external person understand you meant the v0.21 of MeiliSearch (the core-engine)

Comment (fharper) [reply opensource.md:9]: Oh, ok, make sense, I will change it, thanks

**Review by mdubus — APPROVED** [review]

**Review by bidoubiwa — APPROVED** [review]

**Comment by fharper** [comment]
@react-learner can you review this PR please? We'll merge and I'll add the suggestions in new PRs :)

**Comment by fharper** [comment]
@react-learner: a friendly reminder :D

**Review by MarinPostma — APPROVED** [review]

**Pull request merged**